server/datastore/mysql/policies.go

```diff
@@ -11,18 +10,35 @@ import (
 	"github.com/fleetdm/fleet/v4/server/contexts/ctxerr"
 	"github.com/fleetdm/fleet/v4/server/fleet"
 	"github.com/jmoiron/sqlx"
+	"github.com/pkg/errors"
 )
 
-func (ds *Datastore) NewGlobalPolicy(ctx context.Context, queryID uint, resolution string) (*fleet.Policy, error) {
-	res, err := ds.writer.ExecContext(ctx, `INSERT INTO policies (query_id, resolution) VALUES (?, ?)`, queryID, resolution)
-	if err != nil {
+func (ds *Datastore) NewGlobalPolicy(ctx context.Context, authorID, queryID uint, name, query, description, resolution string) (*fleet.Policy, error) {
```

Comment:
We usually use `*uint` and `nil` references the case where you are not receiving `queryID`.

It's mostly a nit, because ids start at 1 iirc, but it'd be more consistent if you used a pointer, IMO.